plugins/parodos/src/components/notification/NotificationsList.tsx

```diff
@@ -0,0 +1,276 @@
+import React, {
+  type MouseEvent,
+  type ChangeEvent,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import {
+  Dialog,
+  Grid,
+  makeStyles,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TablePagination,
+  TableRow,
+} from '@material-ui/core';
+import type { NotificationsSlice, NotificationState } from '../../stores/types';
+import { NotificationContent } from '../../models/notification';
+import { errorApiRef, type FetchApi, useApi } from '@backstage/core-plugin-api';
+import { NotificationListItem } from './NotificationsListItem';
+import { NotificationListHeader } from './NotificationsListHeader';
+import { Confirm } from './Confirm';
+import { Progress, SelectedItems } from '@backstage/core-components';
+import { assert } from 'assert-ts';
+import { Alert } from '@material-ui/lab';
+import DialogContent from '@material-ui/core/DialogContent';
+
+const useStyles = makeStyles(theme => ({
+  container: {
+    maxWidth: theme.breakpoints.values.xl,
+  },
+  root: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+  tbody: {
+    '& tr:last-child': {
+      borderBottom: `1px solid ${theme.palette.grey.A100}`,
+    },
+  },
+}));
+
+type Action = 'DELETE' | 'ARCHIVE';
+
+type NotificationListProps = Pick<
+  NotificationsSlice,
+  | 'notifications'
+  | 'fetchNotifications'
+  | 'deleteNotification'
+  | 'notificationsLoading'
+  | 'setNotificationState'
+> & { fetch: FetchApi['fetch'] };
+
+export function NotificationList({
+  notifications,
+  fetchNotifications,
+  deleteNotification,
+  notificationsLoading: loading,
+  setNotificationState,
+  fetch,
+}: NotificationListProps): JSX.Element {
+  const styles = useStyles();
+  const [selectedNotifications, setSelectedNotifications] = useState<
+    NotificationContent[]
```

Comment:
Is there a reason to store a whole notification rather than its id?

Reply:
Yes, because we use the notification to render the table cells, e.g.

```ts
<TableCell component="th" scope="row">
  {notification.subject}
</TableCell>
<TableCell component="th">
  {getHumanReadableDate(notification.createdOn)}
</TableCell>
```

Reply:
changed